salt/pillar/s3.py

```diff
@@ -337,7 +337,7 @@ def __get_pillar_environments(files):
 
     log.debug('Writing S3 buckets pillar cache file')
 
-    with salt.utils.fopen(cache_file, 'w') as fp_:
+    with salt.utils.fopen(cache_file, 'wb') as fp_:
```

Comment:
this should be `'wb' if six.PY3 else 'w'`

and six imported like `import salt.ext.six as six`

Reply:
@JamesSnow putting it all together:
```python
import salt.ext.six as six
with salt.utils.fopen(cache_file, 'wb' if six.PY3 else 'w') as fp_:
   ...
```